ClinGen/clincoded
# Connects to #690. Variant interpretation forms

### Implemented changes:

* Universal evaluation forms for VCI
  * Creates one evaluation per criteria code, and updates interpretation object as needed
  * Form supports dropdowns and checkboxes for criteria value, and textarea fields for criteria description
  * Creates evaluation evidence object if applicable
  * Evaluation evidence (optional) linked to evaluation objects
  * Supports multiple evaluations/criteria codes per form group (Save button)
  * All evaluation objects grouped in one form group share the same evaluation evidence object (if applicable)
* VCI `index` loads in full interpretation object, not just the UUID
* Interpretation tabs now have `forceRenderTabPanel` enabled, which makes them load external data sources in the background; faster/more seamless switching between tabs

### Screenshots:
![image](https://cloud.githubusercontent.com/assets/4326866/16210564/7d730f04-36f1-11e6-8b13-e990f2511f6d.png)

![image](https://cloud.githubusercontent.com/assets/4326866/16210414/afe17d00-36f0-11e6-98d2-9f5168dc9dbc.png)

### Testing:

1. Select a variant to curation in the VCI
2. Add interpretation
3. Update and save forms in Population and Predictors tabs
4. Make sure that buttons and feedback messages next to buttons work properly
5. In DB, confirm that changes to interpretation objects are correct:
  * Interpretation should link to proper evaluation objects
  * One evaluation object per criteria code that was saved
  * One evaluation evidence object per evaluation object, and if multiple evaluations were tied to one save button, they should all point to the same evaluation evidence object
  * Note: evaluations generated in Predictors tab does not have any evaluation evidence objects associated with them, so they should not be linked to one
